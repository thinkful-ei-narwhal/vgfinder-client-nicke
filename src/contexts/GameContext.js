import React, { Component } from "react";

export const nullGame = {
  id: "",
  title: "",
  description: "",
  genre: "",
  rating: "",
  release_date: "",
  developer: "",
  trailer_url: "",
  image_url_box_art: "",
  image_url_two: "",
  image_url_three: "",
  image_url_four: "",
  image_url_five: "",
};

const GameContext = React.createContext({
  game: nullGame,
  setGame: () => {},
  clearGame: () => {},
});

export default GameContext;

export class GameProvider extends Component {
  state = {
    game: nullGame,
  };

  setGame = (game) => {
    this.setState({ game });
  };

  clearGame = () => {
    this.setGame(nullGame);
  };

  render() {
    const value = {
      game: this.state.game,
      setGame: this.setGame,
      clearGame: this.clearGame,
    };
    return (
      <GameContext.Provider value={value}>
        {this.props.children}
      </GameContext.Provider>
    );
  }
}
