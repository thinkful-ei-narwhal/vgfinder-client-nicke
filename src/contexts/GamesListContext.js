import React, { Component } from 'react';

const GamesListContext = React.createContext({
  gamesList: [],
  error: null,
  setError: () => { },
  clearError: () => { },
  setGamesList: () => { },
})
export default GamesListContext

export class GamesListProvider extends Component {
  state = {
    gamesList: [],
    error: null,
  };

  setGamesList = gamesList => {
    this.setState({ gamesList })
  }

  setError = error => {
    console.error(error)
    this.setState({ error })
  }

  clearError = () => {
    this.setState({ error: null })
  }

  render() {
    const value = {
      gamesList: this.state.gamesList,
      error: this.state.error,
      setError: this.setError,
      clearError: this.clearError,
      setGamesList: this.setGamesList,
    }
    return (
      <GamesListContext.Provider value={value}>
        {this.props.children}
      </GamesListContext.Provider>
    )
  }
}
