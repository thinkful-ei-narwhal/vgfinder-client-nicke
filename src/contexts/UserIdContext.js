import React, { Component } from 'react'

const UserIdContext = React.createContext({
  userId: undefined,
  wishlist: [],
  setUserId: () => { },
  setUserWishlist: () => { },
})

export default UserIdContext

export class UserIdContextProvider extends Component {
  state = {
    userId: undefined,
    wishlist: []
  };

  setUserId = (userId) => {
    this.setState({ userId })
  }

  setUserWishlist = (wishlist) => {
    this.setState({ wishlist })
  }


  render() {
    const value = {
      userId: this.state.userId,
      wishlist: this.state.wishlist,
      setUserId: this.setUserId,
      setUserWishlist: this.setUserWishlist
    }

    return (
      <UserIdContext.Provider value={value}>
        {this.props.children}
      </UserIdContext.Provider>
    )
  }
}
