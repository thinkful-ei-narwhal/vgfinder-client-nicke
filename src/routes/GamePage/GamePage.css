.trailer-section {
  margin-top: 250px;
  margin-bottom: 250px;
}
