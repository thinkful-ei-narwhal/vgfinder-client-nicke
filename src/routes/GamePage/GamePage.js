import React, { Component } from 'react';
import GamesApiService from '../../services/games-api-service';
import { Section } from '../../components/Utils/Utils';
import Header from './../../components/Header/Header';
import './GamePage.css'
import GamesCarousel from '../../components/GameCarousel/GameCarousel';
import GameContext from '../../contexts/GameContext'
import GameInfo from '../../components/GameInfo/GameInfo';
import YouTube from 'react-youtube';

export default class GamesHomePage extends Component {
  static contextType = GameContext;

  constructor(props) {
    super(props)
    this.state = {
      loaded: false
    }
  }

  componentDidMount() {
    const { gameId } = this.props.match.params;
    this.context.clearError();
    GamesApiService.getGame(gameId)
      .then(res => {
        this.context.setGame(res);
        this.setState({ loaded: true });
      })
      .catch(this.context.setError)
  }

  componentWillUnmount() {
    this.context.clearGame();
  }

  renderGame() {
    const { error } = this.context

    const opts = {
      height: '390',
      width: '640',
      playerVars: {
        autoplay: 0,
      },
    };

    //Running loop twice to garauntee box art is first item pushed
    const reel = [];
    for (const [key, value] of Object.entries(this.context.game)) {
      if (key === "image_url_box_art") {
        reel.push({ imgUrl: value, gameId: this.context.game.id });
      }
    }
    for (const [key, value] of Object.entries(this.context.game)) {
      if (key !== "image_url_box_art" && key.includes("image") && value) {
        reel.push({ imgUrl: value, gameId: this.context.game.id });
      }
    }

    const youtubeUrl = this.context.game.trailer_url.split("=");
    const youtubeId = youtubeUrl[1].substring(0, youtubeUrl[1].indexOf('&'))

    return <>
      <Header pathName={this.props.location.pathname} />
      <Section className='GamePage'>
        <h2>{this.context.game.title}</h2>
        {error ? <p className='red'>There was an error, try again</p> : <GamesCarousel reel={reel} isSingleGame={true} />}
        <GameInfo game={this.context.game} />
      </Section>
      <Section>
        <h2>Trailer</h2>
        <YouTube videoId={youtubeId} opts={opts} onReady={this._onReady} />
      </Section>
    </>
  }

  render() {
    return (
      <>
        {this.state.loaded ? this.renderGame() : null}
      </>
    )
  }
}