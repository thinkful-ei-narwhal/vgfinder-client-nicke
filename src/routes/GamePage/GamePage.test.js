import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import GamePage from './GamePage'
import { GameProvider } from './../../contexts/GameContext'

describe(`Game Page component`, () => {
  it('renders a Game Page component', () => {
    const wrapper = shallow(
      <GameProvider>
        <GamePage />
      </GameProvider>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
