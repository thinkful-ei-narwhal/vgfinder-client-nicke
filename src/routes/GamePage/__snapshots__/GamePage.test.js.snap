// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Page component renders a Game Page component 1`] = `
<ContextProvider
  value={
    Object {
      "clearError": [Function],
      "clearGame": [Function],
      "error": null,
      "game": Object {
        "description": "",
        "developer": "",
        "genre": "",
        "id": "",
        "image_url_box_art": "",
        "image_url_five": "",
        "image_url_four": "",
        "image_url_three": "",
        "image_url_two": "",
        "rating": "",
        "release_date": "",
        "title": "",
        "trailer_url": "",
      },
      "setError": [Function],
      "setGame": [Function],
    }
  }
>
  <GamesHomePage />
</ContextProvider>
`;
